### PR TITLE
`list-dirs` support and implementation

### PR DESCRIPTION
**Pull request by iomonad**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -156,6 +156,26 @@ examples
...
+
+**arguments**:
+- path
+- [options]
+
+**returns**: seq of string
+
+examples
+```clojure
+(doall (map println (list-dirs "gs://my-bucket/dir/")))
```
Comment (RolT) [inline README.md:171]: Might as well use `run!`

File: src/tools/io.clj [hunk 2]
```diff
@@ -144,6 +144,9 @@
 (def ^{:doc "Alias of tools.io.core/list-files for backward compatibility."
        :arglists '([path & [options]])} list-files tools.io.core/list-files)
 
+(def ^{:doc "Alias of tools.io.core/list-dirs for backward compatibility."
```
Comment (RolT) [inline src/tools/io.clj:147]: The docstring is a bit strange , it's a new function.

File: src/tools/io/core.clj [hunk 3]
```diff
@@ -168,6 +172,20 @@
                                 (str/starts-with? (.getName file) prefix))))
                  (map (fn [^File f] (.getPath f)))))))))
 
+(defmethod list-dirs :base
+  [path & [_options]]
+  (let [^File f (io/file path)]
+    (cond
+      (.isDirectory f)
+        (->> (.listFiles f)
```
Comment (RolT) [inline src/tools/io/core.clj:180]: The indentation is off

File: src/tools/io/core.clj [hunk 4]
```diff
@@ -168,6 +172,20 @@
...
+             (filter #(.isDirectory ^File %))
+             (map #(.getPath ^File %)))
+
+       ;; trailing slash but not a directory: nil
+       (str/ends-with? path File/separator)
+       nil
+       :else nil)))
```
Comment (RolT) [inline src/tools/io/core.clj:187]: I'm not sure if those 2 last conditions provide anything. `cond` with no matching condition is `nil` anyway. You could remove those two conditions, and even replace the `cond` by `when`

**Review by RolT — COMMENTED** [review]
Thanks for the PR, I added a few comments

**Comment by nikopol** [comment]
everything is ok, except that you should do your merge request to the devel branch, without modifying the version number.

**Pull request closed without merging**